cellmlmanip/model.py

```diff
@@ -410,27 +410,46 @@ def get_equation_graph(self, refresh=False) -> nx.DiGraph:
         self.graph = graph
         return graph
 
-    def get_equations_for(self, symbols, lexicographical_sort=True):
+    def get_equations_for(self, symbols, sort_by_input_symbols=False, excluded_symbols=[]):
         """Get all equations for given collection of symbols
 
         Results are sorted in topographical order.
         :param symbols: the symbols to get the equations for
-        :param lexicographical_sort: indicates whether the result is sorted in lexicographical order first
+        :param sort_by_input_symbols: indicates whether the sorting in symbols determines the sorting of the output
+        :param excluded_symbols: list of symbols to exlcude
         """
         graph = self.get_equation_graph()
-        if lexicographical_sort:
-            sorted_symbols = nx.lexicographical_topological_sort(graph, key=str)
-        else:
-            sorted_symbols = nx.topological_sort(graph)
-
-        # Create set of symbols for which we require equations
-        required_symbols = set()
+        sorted_symbols = [s for s in nx.topological_sort(graph)]
+
+        def get_parents(node, graph=graph, sorted_symbols=sorted_symbols):
```

Comment:
I'd like this at least to have a docstring, even if we don't make it a class method.